### PR TITLE
fix: node RPCCall returns nil instead of error

### PR DESCRIPTION
**Pull request by vidarandrebo**
Fixed a bug where an error was not returned from node.RPCCall
Added tests for all return paths in RPCCall function.

### CODE REVIEW

File: rpc.go [hunk 1]
```diff
@@ -18,18 +18,18 @@ type CallData struct {
...
-func (n *RawNode) RPCCall(ctx context.Context, d CallData) (resp protoreflect.ProtoMessage, err error) {
+func (n *RawNode) RPCCall(ctx context.Context, d CallData) (protoreflect.ProtoMessage, error) {
 	md := &ordering.Metadata{MessageID: n.mgr.getMsgID(), Method: d.Method}
 	replyChan := make(chan response, 1)
 	n.channel.enqueue(request{ctx: ctx, msg: &Message{Metadata: md, Message: d.Message}}, replyChan, false)
 
 	select {
 	case r := <-replyChan:
 		if r.err != nil {
-			return nil, err
```
Comment (vidarandrebo) [inline rpc.go:29]: err is always nil here

File: rpc.go [hunk 2]
```diff
@@ -18,18 +18,18 @@ type CallData struct {
...
+			return nil, r.err
 		}
 		return r.msg, nil
 	case <-ctx.Done():
-		return resp, ctx.Err()
```
Comment (vidarandrebo) [inline rpc.go:33]: resp is always nil here

File: rpc.go [hunk 3]
```diff
@@ -18,18 +18,18 @@ type CallData struct {
...
 // This method should be used by generated code only.
-func (n *RawNode) RPCCall(ctx context.Context, d CallData) (resp protoreflect.ProtoMessage, err error) {
+func (n *RawNode) RPCCall(ctx context.Context, d CallData) (protoreflect.ProtoMessage, error) {
 	md := &ordering.Metadata{MessageID: n.mgr.getMsgID(), Method: d.Method}
 	replyChan := make(chan response, 1)
 	n.channel.enqueue(request{ctx: ctx, msg: &Message{Metadata: md, Message: d.Message}}, replyChan, false)
 
 	select {
 	case r := <-replyChan:
 		if r.err != nil {
```
Comment (meling) [inline rpc.go:28]: Here we can simply return without the proceeding `if`:
```go
return r.msg, r.err
```
Or did I miss something? The tests still pass with this version.

File: rpc_test.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+	}
+}
+
+func initServer() *gorums.Server {
+	srv := gorums.NewServer()
+	dummy.RegisterDummyServer(srv, &testSrv{})
+	return srv
+}
+
+func rpcTestMgr() *dummy.Manager {
```
Comment (meling) [inline rpc_test.go:106]: Maybe a better name for this helper would be `gorumsTestMgr` or something?

**Review by meling — CHANGES_REQUESTED** [review]
Looks great! A few minor things.

**Review by meling — APPROVED** [review]

**Pull request merged**